backend/routes.js

```diff
@@ -666,7 +709,7 @@ route.GET('/kv/{contractID}/{key}', {
 }, async function (request, h) {
   const { contractID, key } = request.params
 
-  if (key.startsWith('_private')) {
+  if (!CONTRACT_DATA_REGEX.test(contractID) || !key || key.includes('\x00') || key.startsWith('_private')) {
```

Comment:
Many of these improved checks have been added to this file (not just here), and I'm thinking: since these represent requests that should never be made in the first place, could we maybe return a different error than "not found"? E.g. bad request?

Reply:
Hm, not really sure. I think that a 404 is a valid response, and so is 400, depending on perspective. A 404 is valid because 'technically' there could be a response but we know there won't be / there shouldn't be.

A 400 is also valid. The spec says: 'The request could not be understood by the server due to malformed syntax.', and we could say that the path is itself invalid and there couldn't possibly be a response.

Let me know if I should change this to 400.

Reply:
yes, 400 (or something similar) would be better than (not found) IMO

Reply:
Made this change. However, personally I prefer a 404 in these less clear cases as it divulges slightly less information to clients.

Reply:
yeah but ... it's not like we're hiding anything important.